pycqed/measurement/waveform_control/fluxpulse_predistortion.py

```diff
@@ -128,37 +135,53 @@ def convert_expmod_to_IIR(expmod, dt, inverse_IIR=True):
         filter is returned in the form [a, b].
     """
     if hasattr(expmod[0], '__iter__'):
+        # FIXME: probably this does not work for multiple sos filter lists
         iir = [convert_expmod_to_IIR(e, dt, inverse_IIR) for e in expmod]
         a = [i[0] for i in iir]
         b = [i[1] for i in iir]
     else:
         A, B, tau = expmod
         if np.array(tau).ndim > 0:  # sum of exp mod
-            import sympy
+            import sympy as sp
             N = len(tau)
-            a = sympy.symbols(','.join([f'a{i}' for i in range(N + 1)]))
-            tau_s = sympy.symbols(','.join([f'tau{i + 1}' for i in range(
-                N)]))
-            if N == 1:
-                tau_s = [tau_s]
-            T = sympy.symbols('T')
-            z = sympy.symbols('z')
-            p = 2 / T * (1 - 1 / z) / (1 + 1 / z)
-            r = [sympy.prod(
-                [tau * p + 1 for i, tau in enumerate(tau_s) if i != j]) for j
-                in
-                range(N)]
-            s = sympy.prod([tau * p + 1 for tau in tau_s])
-            f = (a[0] * s + sum(
-                [r * tau * a * p for r, tau, a in zip(r, tau_s, a[1:])])) / s
-
-            n, d = sympy.fraction(f.simplify())
-            sym_coeffs_n = n.as_poly(z).all_coeffs()
-            coeffs_n = sympy.lambdify([a] + [tau_s] + [T], sym_coeffs_n)
-            sym_coeffs_d = d.as_poly(z).all_coeffs()
-            coeffs_d = sympy.lambdify([a] + [tau_s] + [T], sym_coeffs_d)
-            b = np.array(coeffs_d([A] + B, tau, dt))
-            a = np.array(coeffs_n([A] + B, tau, dt))
+            a = [sp.Rational(a) for a in [A] + list(B)]
+            tau_s = [sp.Rational(t / 1e-9) * 1e-9 for t in list(tau)]
+            # In the next line, going via the reciprocal value is more precise
+            # since we usually specify dt a 1 over sampling frequency.
```

Comment:
small typo, should be:
_since we usually specify dt as 1 over sampling frequency._